### DISCUSSION TITLE
Joe's script to the rescue - tox tests working now

### DISCUSSION THREAD

**Discussion by Elliria** [autokey/autokey#859] [Show and tell]
I've been wrestling with the `tox` tests for quite some time to try to get them working so that the same tests GitHub runs can be run locally. They've been broken for me for quite some time and I had all but given up. But with your help, @josephj11, the problem has been solved and `tox` test are now working.

[Your script](https://github.com/autokey/autokey/issues/847#issuecomment-1545889881) showed me that I was missing the `gir1.2-appindicator3-0.1` and `pyqt5-dev-tools` packages.

After installing those, I tried running the `tox` tests just to see if they were fixed, but I still got an error. It was just a complaint that I didn't have `tkinter` installed. Not a problem. I installed the `python3-tk` file and tried to run the `tox` test again.

It worked. I'm gobsmacked. I've been fighting and fighting with `tox` for months and all of a sudden it works perfectly *(except for a specific couple of the [many tox tests](https://github.com/autokey/autokey/wiki/GitHub-Cheat-Sheet#local---run-tests-on-a-clone) that aren't working for unrelated reasons that will be taken care of elsewhere)*.

Now the question is whether the `python3-tk` file should be added to the [apt-requirements.txt](https://github.com/autokey/autokey/blob/develop/apt-requirements.txt) file or not since it's only needed, in this case, by `tox`, which is a tool used by developers.

I've already added it to the [Local - prepare your computer or virtual machine for development](https://github.com/autokey/autokey/wiki/GitHub-Cheat-Sheet#local---prepare-your-computer-or-virtual-machine-for-development) section of the [GitHub Cheat Sheet](https://github.com/autokey/autokey/wiki/GitHub-Cheat-Sheet) page in the wiki.

Thoughts?

**Comment by josephj11** [comment]
There's a requirements section somewhere that says these things need to be installed if you want to do development (tox...). That's where tk should probably be added. I don't think normal users need it for anything although it is another GUI interaction library that occasionally gets used by AutoKey users to do things that our API calls don't support.

**Reply by Elliria** [reply]
Yep. That's the [CONTRIBUTORS.rst](https://github.com/autokey/autokey/blob/develop/CONTRIBUTORS.rst) file that will be updated by and by.

Also, if you've never played around with Tkinter, you're in for quite a treat if you do. I've got 120 Tkinter scripts from all the experimentation I've done with it because I just couldn't stop.
